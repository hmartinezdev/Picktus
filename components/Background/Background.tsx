import React, { Component } from 'react';

interface PropsType {
  children: JSX.Element[] | JSX.Element;
}

class Background extends Component<PropsType, {}> {
  render() {
    const { children } = this.props;

    return (
      <div className="background">
        {children}
        <style jsx>{`
          .background {
            display: flex;
            align-self: stretch;
            min-height: 100vh;
            width: 100%;
            background: url('https://source.unsplash.com/collection/1922729/1920x1080') no-repeat center center fixed;
            background-size: cover;
          }
        `}</style>
      </div>
    );
  }
}

export default Background;
