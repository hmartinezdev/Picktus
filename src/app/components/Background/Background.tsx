import React, { Component } from 'react';

interface IBackgroundPropsType {
  children: JSX.Element[] | JSX.Element;
}

class Background extends Component<IBackgroundPropsType, {}> {
  public render(): React.ReactElement<Background> {
    const { children } = this.props;

    return (
      <div className="background">
        {children}
        <style jsx>{`
          .background {
            display: flex;
            align-self: stretch;
            min-height: 100vh;
            width: 100%;
            background: url('/static/background.svg') no-repeat center center fixed;
            background-size: cover;
          }
        `}</style>
      </div>
    );
  }
}

export default Background;
