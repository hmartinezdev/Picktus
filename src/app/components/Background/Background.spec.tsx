import { shallow } from 'enzyme';
import React from 'react';
import Background from './Background';

let wrapper;

const setup = () =>
  shallow(
    <Background>
      <div />
    </Background>
  );

describe('<Background />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
