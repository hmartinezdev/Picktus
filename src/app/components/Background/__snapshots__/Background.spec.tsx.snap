// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> should render properly 1`] = `
<div
  className="background"
>
  <div />
  <style
    jsx={true}
  >
    
          .background {
            display: flex;
            align-self: stretch;
            min-height: 100vh;
            width: 100%;
            background: url('/static/background.svg') no-repeat center center fixed;
            background-size: cover;
          }
        
  </style>
</div>
`;
